cornell-dti/course-plan
# Opt in/out Frontend

### Summary 






This PR implements the frontend for the opt in/out modal. This includes two parts:
- [x] Task 1: Add the Settings icon with the Menu with “Replace, Remove” and “Replace” menus in the Req Bar. The Settings icon should be gray, but blue when hovered over (these colors already exist in the code base)
- [x] Task 2: Create the Replace and Remove modals




### Test Plan 


To test the functionality, I will
- Hover over the gear icon to ensure that the gear changes colors
- Click on the gear icon to ensure the modal opens
- Click on the remove button to open the remove modal
- Click on the replace button to op
- - Choose a class not in the user's current schedule and one that is

For a class that is selected, there are three cases:
1. The class is not in the schedule, we need to add it
2. The class is already in the schedule once, so we just close the modal
3. The class is in the schedule more than once, so we open a modal to resolve duplicates

Case with 0 of that class
<img width="473" alt="Screen Shot 2022-09-14 at 9 01 45 PM" src="https://user-images.githubusercontent.com/53061040/190289018-2c50cd52-b075-4594-9098-ccbd04980363.png">
Case with 2+ of that class
<img width="492" alt="Screen Shot 2022-09-14 at 9 01 23 PM" src="https://user-images.githubusercontent.com/53061040/190289019-253d0a40-5003-4aba-892b-4f1403aecdcd.png">
